sdk/storage/azure-storage-blob/azure/storage/blob/_models.py

```diff
@@ -508,6 +552,17 @@ def __init__(self, **kwargs):
         self.encryption_scope = kwargs.get('x-ms-encryption-scope')
         self.request_server_encrypted = kwargs.get('x-ms-server-encrypted')
 
+        # these are for delete_blobs settings
+        self.delete_snapshots = kwargs.pop('delete_snapshots', None)
+        self.lease_id = kwargs.pop('lease_id', None)
+        self.if_modified_since = kwargs.pop('if_modified_since', None)
+        self.if_unmodified_since = kwargs.pop('if_unmodified_since', None)
+        self.match_condition = kwargs.pop('match_condition', None)
+        self.timeout = kwargs.pop('timeout', None)
```

Comment:
We should removed these from the public BlobProperties API.
These are parameters applied as input only to batch operations

Reply:
discussed offline, the blob type could be a dict, existing user can also treat BlobProperties as a dict.